internal/gatewayapi/translator.go

```diff
@@ -310,9 +310,11 @@ func (t *Translator) InitIRs(gateways []*GatewayContext) (map[string]*ir.Xds, ma
 		if t.MergeGateways {
 			maps.Copy(labels, GatewayClassOwnerLabel(string(t.GatewayClassName)))
 			gwInfraIR.Proxy.GetProxyMetadata().Labels = labels
+			gwInfraIR.Proxy.GetProxyMetadata().Name = irKey
```

Comment:
should this logic be ?
if GatewayNamespaceMode {
gwInfraIR.Proxy.GetProxyMetadata().OwnerReference = &ir.ResourceMetadata {
Kind: "Gateway"
......
} else {

gwInfraIR.Proxy.GetProxyMetadata().OwnerReference = &ir.ResourceMetadata {
Kind: Deployment
Name: ....
}

Reply:
I would also be up for setting OwnerReference, having two Names fields in InfraIR is somewhat confusing